### ISSUE TITLE
Profile fetch doesn't seem to have correct caching behaviour

### ISSUE THREAD

**Issue by rfk** [mozilla/application-services#2615]
I instrumented a local Fenix build to log all network activity, and as expected, I can see it periodically fetching my profile info from the FxA servers:

```
02-11 14:55:46.446 19552 19614 I FirefoxAccountStateMach: Fetching profile...
02-11 14:55:46.446 19552 19583 D fxaclient_ffi: fxa_profile
02-11 14:55:46.446 19552 19583 W GeckoViewFetchClient: GV FETCH GET https://profile.accounts.firefox.com/v1/profile
02-11 14:55:47.156 19552 19583 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://profile.accounts.firefox.com/v1/profile 200
...
02-11 15:09:48.286 23405 23466 I FirefoxAccountStateMach: Fetching profile...
02-11 15:09:48.286 23405 23436 D fxaclient_ffi: fxa_profile
02-11 15:09:48.286 23405 23436 W GeckoViewFetchClient: GV FETCH GET https://profile.accounts.firefox.com/v1/profile
02-11 15:09:48.896 23405 23436 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://profile.accounts.firefox.com/v1/profile 200
...
02-11 15:51:25.226 31881 31943 I FirefoxAccountStateMach: Fetching profile...
02-11 15:51:25.236 31881 31925 D fxaclient_ffi: fxa_profile
02-11 15:51:25.236 31881 31925 W GeckoViewFetchClient: GV FETCH GET https://profile.accounts.firefox.com/v1/profile
02-11 15:51:25.856 31881 31925 W GeckoViewFetchClient: GV FETCH RESPONSE GET https://profile.accounts.firefox.com/v1/profile 200
```

What's _not_ expected is that each of these fetches of my profile data is responding with a `200 OK` status code. I haven't changed any of the profile information on this account, so after the first profile fetch, I would expect subsequent requests to be sending an etag and receiving a `304 Not Modified` response.

I haven't dug into why. Perhaps our elaborate networking facade is dropping the etag header somewhere along the way? Perhaps there's an etag-related bug in the FxA server?

Non-urgent, but we should figure out what's going on here for completeness.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-2444)
┆Epic: Moving fxa-client functionality out of android-components
┆Sprint End Date: 2021-06-25

**Comment by data-sync-user** [comment]
➤ Tarik Eshaq commented:

Did some further investigation but going to leave this back in the backlog for later.

Here’s what I discovered:
I added logs at the lowest level of the rust layer in our fxa-client component and the etag seems to still be there. The etag seemed to be formated with some escape slashes.

This leaves us with one of the following causes:

  * It could be a bug with the FxA server, where it doesn’t properly handle the correct etag
  * It could be a bug with the viaduct implementation in fenix, specifically the thin layer in android components - I doubt this is a problem with GeckoView but I wouldn’t write it off
  * It could be a bug in how the etag is formated, when I looked at it, it was accompanied with escaping slashes, so that’s a possible cause.

Anyways, will put this back in the backlog, hopefully, the above context can make any further investigation easier!
